hbase-client/src/main/java/org/apache/hadoop/hbase/filter/MultipleColumnPrefixFilter.java

```diff
@@ -40,15 +42,18 @@
  */
 @InterfaceAudience.Public
 public class MultipleColumnPrefixFilter extends FilterBase {
+  private static final Logger LOG = LoggerFactory.getLogger(MultipleColumnPrefixFilter.class);
   protected byte [] hint = null;
   protected TreeSet<byte []> sortedPrefixes = createTreeSet();
   private final static int MAX_LOG_PREFIXES = 5;
 
   public MultipleColumnPrefixFilter(final byte [][] prefixes) {
     if (prefixes != null) {
       for (int i = 0; i < prefixes.length; i++) {
-        if (!sortedPrefixes.add(prefixes[i]))
-          throw new IllegalArgumentException ("prefixes must be distinct");
+        if (!sortedPrefixes.add(prefixes[i])) {
+          LOG.error("prefix {} is repeated", Bytes.toString(prefixes[i]));
+          throw new IllegalArgumentException("prefixes must be distinct");
```

Comment:
Now that you are doing this improvement, maybe you can improve looping also:
```
      for (byte[] prefix : prefixes) {
        if (!sortedPrefixes.add(prefix)){
        }
      }

```

Reply:
Fixed.